webpack/components/extensions/Hosts/BulkActions/BulkPackagesWizard/02_BulkPackagesTable.js

```diff
@@ -0,0 +1,155 @@
+  TextVariants,
+} from '@patternfly/react-core';
+import TableIndexPage from 'foremanReact/components/PF4/TableIndexPage/TableIndexPage';
+import { translate as __ } from 'foremanReact/common/I18n';
+import SelectAllCheckbox from 'foremanReact/components/PF4/TableIndexPage/Table/SelectAllCheckbox';
+import { getControllerSearchProps } from 'foremanReact/constants';
+import { noop } from 'foremanReact/common/helpers';
+import { RowSelectTd } from 'foremanReact/components/HostsIndex/RowSelectTd';
+import { getPageStats } from 'foremanReact/components/PF4/TableIndexPage/Table/helpers';
+import { BulkPackagesWizardContext, getPackagesUrl } from './BulkPackagesWizard';
+import katelloApi from '../../../../../services/api';
+
+export const BulkPackagesUpgradeTable = props => <BulkPackagesTable {...props} tableType="upgrade" />;
+export const BulkPackagesInstallTable = props => <BulkPackagesTable {...props} tableType="install" />;
+
+const BulkPackagesTable = ({
+  tableType,
+}) => {
+  const {
+    setShouldValidateStep2,
+    packagesBulkSelect,
+    packagesResults: results,
+    packagesMetadata: {
+      total, per_page: perPage, page, subtotal,
+    },
+    packagesResponse: response,
+  } = useContext(BulkPackagesWizardContext);
+  const PACKAGES_URL = getPackagesUrl(tableType);
+  const apiOptions = { key: 'BULK_HOST_PACKAGES' };
+
+  const origSearchProps = getControllerSearchProps('packages', 'searchBar-packages');
+  const customSearchProps = {
+    ...origSearchProps,
+    autocomplete: {
+      ...origSearchProps.autocomplete,
+      url: katelloApi.getApiUrl('/packages/auto_complete_search'),
+    },
+  };
+
+  const {
+    selectPage,
+    selectNone,
+    selectOne,
+    isSelected,
+    selectedCount,
+    areAllRowsSelected,
+    areAllRowsOnPageSelected,
+    updateSearchQuery,
+    hasInteracted,
+  } = packagesBulkSelect;
+
+  useEffect(() => {
+    if (results?.length && hasInteracted) {
+      setShouldValidateStep2(true);
+    }
+  }, [setShouldValidateStep2, results, hasInteracted]);
+
+  const pageStats = getPageStats({ total: subtotal, page, perPage });
+  const selectionToolbar = (
+    <ToolbarItem key="selectAll">
+      <SelectAllCheckbox
+        {...{
+          selectNone,
+          selectPage,
+          selectedCount,
+          pageRowCount: pageStats.pageRowCount,
+          areAllRowsSelected,
+          areAllRowsOnPageSelected,
+        }}
+        selectAll={noop}
+        totalCount={total}
+        areAllRowsOnPageSelected={areAllRowsOnPageSelected()}
+        areAllRowsSelected={areAllRowsSelected()}
+      />
+    </ToolbarItem>
+  );
+
+  const columns = {
+    name: {
+      title: __('Package'),
+      wrapper: ({ name, id }) => (
+        <a target="_blank" href={`/packages/${id}`} rel="noreferrer">{name}</a>
+      ),
+      isSorted: true,
+      weight: 50,
+    },
+  };
+
+  // to pass in so that TableIndexPage doesn't alter browser URL
+  const restrictedSearchQuery = query => query;
```

Comment:
Is this something other components will need to do?

Reply:
Only components with "embedded" tables (in other words tables that you need to not affect the browser URL params).

Reply:
Should it be added as code/comment in the foreman?

Reply:
Realized that it makes more sense to just use the `updateParamsByUrl` param which was already there. Updated here and in Foreman.